src/components/CodeBlock/index.tsx

```diff
@@ -71,24 +75,48 @@ export const CodeBlock = (props: CodeBlockProps) => {
         }
     }
 
+    const copyToClipboard = () => {
+        if (navigator.clipboard) {
+            navigator.clipboard.writeText(code || props.children.props.children.trim())
+            setHasBeenCopied(true)
+            setTooltipVisible(true)
+            setTimeout(() => {
+                setTooltipVisible(false)
+            }, 1000)
+        }
+    }
+
     return (
-        <Highlight
-            {...defaultProps}
-            code={code || props.children.props.children.trim()}
-            language={language as Language}
-            theme={theme}
-        >
-            {({ className, style, tokens, getLineProps, getTokenProps }) => (
-                <pre className={className} style={{ ...style, padding: '20px' }} id={codeBlockId}>
-                    {tokens.map((line, i) => (
-                        <div key={i} {...getLineProps({ line, key: i })}>
-                            {line.map((token, key) => (
-                                <span key={key} {...getTokenProps({ token, key })} />
-                            ))}
-                        </div>
-                    ))}
-                </pre>
-            )}
-        </Highlight>
+        <>
+            <Highlight
+                {...defaultProps}
+                code={code || props.children.props.children.trim()}
+                language={language as Language}
+                theme={theme}
+            >
+                {({ className, style, tokens, getLineProps, getTokenProps }) => (
+                    <pre className={className} style={{ ...style, padding: '20px' }} id={codeBlockId}>
+                        <Tooltip title="Copied!" visible={tooltipVisible}>
+                            <CopyOutlined
+                                style={{
+                                    position: 'absolute',
+                                    right: 15,
+                                    color: hasBeenCopied ? '#9c9a9a' : '#fff',
+                                    transitionDuration: '1s',
+                                }}
+                                onClick={copyToClipboard}
+                            />
+                        </Tooltip>
+                        {tokens.map((line, i) => (
+                            <div key={i} {...getLineProps({ line, key: i })}>
+                                {line.map((token, key) => (
+                                    <span key={key} {...getTokenProps({ token, key })} />
+                                ))}
+                            </div>
+                        ))}
+                    </pre>
+                )}
+            </Highlight>
+        </>
```

Comment:
Why wrapped in a React fragment?

Reply:
lingered from another change I was making, will remove